sbi/inference/snpe/snpe_base.py

```diff
@@ -299,9 +304,15 @@ def train(
 
             self.epoch += 1
 
+
+            log_prob_sum /= len(train_loader) * train_loader.batch_size
+            self._summary["train_log_probs"].append(log_prob_sum)
+
             # Calculate validation performance.
-            self._neural_net.eval()
+            self._neural_net.eval() #<--- this is not required as the validation is
+            # performed under no_grad context manager
```

Comment:
Where did you find this? From reading [this](https://discuss.pytorch.org/t/model-eval-vs-with-torch-no-grad/19615/2), I do think that `.eval()` is still required.

Reply:
Interesting. I picked this up from [here](https://github.com/mackelab/sbi/blob/c9174e130cc622ae2219433dda541b74b37912a3/sbi/inference/snpe/snpe_base.py#L303). Thanks for pointing me at this [explanation](https://discuss.pytorch.org/t/model-eval-vs-with-torch-no-grad/19615/2). I see your point that `self._neural_net.eval()` might contain dropout layers which should go into `eval` model after this line irrespective of autograd being in effect or not. 

I need to search API ref where this is documented besides this forum. Removed with fb83af8